### PR TITLE
better error handling for outdated gitpython module

### PR DESCRIPTION
**Pull request by Caylo**
ref #1757

### CODE REVIEW

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -896,7 +896,10 @@ def check_github():
         else:
             check_res = "OK"
     elif github_user:
-        check_res = "FAIL (unexpected exception: %s)" % push_err
+        if sys.modules['git'].__version__ < "1.0.0":
```
Comment (boegel) [inline easybuild/tools/github.py]: please use `LooseVersion` when comparing versions

also, just use `git.__version__`?

```
if LooseVersion(git.__version__) < LooseVersion('1.0'):
```

Did you verify whether things (should) work with GitPython 1.0 and up?

Comment (Caylo) [inline easybuild/tools/github.py]: Yup, verified that. Everything above 1.0 works.

File: easybuild/tools/github.py [hunk 2]
```diff
@@ -896,7 +898,10 @@ def check_github():
         else:
             check_res = "OK"
     elif github_user:
-        check_res = "FAIL (unexpected exception: %s)" % push_err
+        if LooseVersion(git.__version__) < LooseVersion('1.0'):
+            check_res = "FAIL (outdated gitpython module, try updating first)"
```
Comment (boegel) [inline easybuild/tools/github.py]: can you clarify 'outdated' in the error, i.e. mention what the minimal version is? and also mention the what version was found

something like

```python
check_res = "FAIL (GitPython version %s is too old, should be version %s or newer)" % (git.__version__, req_gp_ver)
```

File: easybuild/tools/github.py [hunk 3]
```diff
@@ -896,7 +898,12 @@ def check_github():
         else:
             check_res = "OK"
     elif github_user:
-        check_res = "FAIL (unexpected exception: %s)" % push_err
+        req_gp_ver = '1.0'
+        if LooseVersion(git.__version__) < LooseVersion(req_gp_ver):
+            check_res = "FAIL (GitPython version %s is too old, should be version %s or newer)" % (git.__version__,
+                         req_gp_ver)
```
Comment (boegel) [inline easybuild/tools/github.py]: bleh on the wrapping here...

maybe like this?

```python
gp_ver, req_gp_ver = git.__version__, '1.0'
if LooseVersion(gp_ver) < LooseVersion(req_gp_ver):
    check_res = "FAIL (GitPython version %s is too old, should be version %s or newer)" % (gp_ver, req_gp_ver)
```

**Comment by boegel** [comment]
Going in, thanks @Caylo!

**Pull request merged**